### ISSUE TITLE
dy operation sometimes creates invalid json (NaN values)

### ISSUE THREAD

**Issue by chowsanthony** [cdisc-org/cdisc-rules-engine#153]
**Describe the problem**
Currently $variables defined in Operations can not be used in the Output Variables section. They can be useful to show users the calculated/dervied values expected.

**Describe the solution**
Allow $variables defined in Operations be used in the Output Variables section.

**Proposed rule logic**
```yaml
Check:
  all:
    - name: --STDY
      operator: not_equal_to
      value: $val_stdy
Operations:
  - name: RFSTDTC
    operator: dy
    domain: DM
    id: $val_stdy
Outcome:
  Message: STDY is not calculated correctly.
  Output Variables:
    - USUBJID
    - RFSTDTC
    - --STDTC
    - $val_stdy
    - --STDY
```

**Comment by nhaydel** [comment]
Does this cause an issue with the rule execution or just a yaml editor error?

**Comment by gcampion-cdisc** [comment]
It looks like this is actually a bug with the dy operation, not an enhancement. It appears that adding $ variables to the output does work. For this particular rule, the engine is returning invalid json:

```json
"errors": [{"value": {"$val_stdy": 1.0}, "row": 1, "uSubjId": "001", "seq": 2}, {"value": {"$val_stdy": NaN}, "row": 2, "uSubjId": "001", "seq": 1}]
```

I've attached the request that triggers the invalid json results:
[cg0222.txt](https://github.com/cdisc-org/cdisc-rules-engine/files/9711684/cg0222.txt)

**Comment by nhaydel** [comment]
This should be fixed with: https://github.com/cdisc-org/cdisc-rules-engine/issues/228
